applications/StructuralMechanicsApplication/custom_constitutive/plasticity_isotropic_kinematic_j2.cpp

```
@@ -0,0 +1,547 @@
+
+//************************************************************************************
+//************************************************************************************
+
+void PlasticityIsotropicKinematicJ2::InitializeMaterialResponseCauchy(ConstitutiveLaw::Parameters& rValues)
+{
+    // Initializations (if any) at each time step
+    
+    // NOTHING DONE HERE
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void PlasticityIsotropicKinematicJ2::CalculateMaterialResponsePK1(ConstitutiveLaw::Parameters& rValues)
+{
+    CalculateMaterialResponseCauchy(rValues); //PK1 not yet implemented
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void PlasticityIsotropicKinematicJ2::CalculateMaterialResponsePK2(ConstitutiveLaw::Parameters& rValues)
+{
+    CalculateMaterialResponseCauchy(rValues); //PK2 not yet implemented
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void PlasticityIsotropicKinematicJ2::CalculateMaterialResponseKirchhoff(ConstitutiveLaw::Parameters& rValues)
+{
+    CalculateMaterialResponseCauchy(rValues); //Kirchhoff not yet implemented
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void PlasticityIsotropicKinematicJ2::CalculateMaterialResponseCauchy(ConstitutiveLaw::Parameters& rValues)
+{
+    BoundedArrayType plastic_strain;
+    double           equivalent_plastic_strain;
+    
+    switch ( VoigtSize )
+    {
+    case 3:
+        //this->CalculateResponse3(rValues, plastic_strain, equivalent_plastic_strain);
```

why this commented?

---

you should throw an error if it is not implemented